app/services/data_migrations/remove_duplicate_provider.rb

```diff
@@ -38,8 +38,15 @@ def change
           courses = permission.training_provider.courses.where(accredited_provider: permission.ratifying_provider)
 
           ActiveRecord::Base.transaction do
+            # For courses run / accredited by duplicate providers, nillify the accredited provider
+            # and remove the accrediting provider id from any associated application choices.
             courses.each do |course|
               course.update!(accredited_provider: nil)
+              choices = ApplicationChoice.where("'#{permission.ratifying_provider.id}' = ANY (provider_ids)")
+              choices.each do |choice|
+                provider_ids = choice.provider_ids - [permission.ratifying_provider.id]
+                choice.update!(provider_ids: provider_ids)
```

Comment:
out of interest how many applications will be touched by this change, as those will be updated in the API syncs for providers?

Reply:
@cwrw 221 application choices would be updated according to prod data today